### DISCUSSION TITLE
Use of JSBSim in c#.net project

### DISCUSSION THREAD

**Discussion by ManvendraSinghDeora** [JSBSim-Team/jsbsim#1095] [General]
I am currently working on a flight simulator project in Unity3D using the JSBSim library to simulate flight physics. As part of this endeavour, I have created a basic wrapper in C# for loading aircraft into the simulation. However, I’ve encountered an issue with sending input from the C# project to JSBSim.

I know the existence of **copy_to_jsbsim** and **copy_from_jsbsim** functions, but I am unsure how to access them in C# within the Unity environment.

Could anyone kindly guide me on how to address these issues? Additionally, I am aware there used to be a wrapper made on an older version of [JSBSim on Sourceforge](https://sourceforge.net/p/jsbsim/mailman/jsbsim-devel/thread/794861.82870.qm%40web35415.mail.mud.yahoo.com/#msg12716571). Still, the link provided in the thread seem to be expired or broken if there already exists a C# wrapper for JSBSim, I would greatly appreciate any information or pointers to relevant resources.

**Comment by seanmcleod** [comment]
I think your best bet would be to build the JSBSim library as a DLL and then for example export **copy_to_jsbsim()** and then use DllImport from C# in order to call it etc.
